python-threatexchange/threatexchange/signal_type/index.py

```diff
@@ -119,15 +120,18 @@ class IndexMatch(t.Generic[T]):
     NamedTuple can't be made generic, so here's an equivalent
     """
 
-    __slots__ = ["distance", "metadata"]
-    distance: int
+    __slots__ = ["match_info", "metadata"]
+    match_info: S_Co
```

Comment:
nit: similarity_info instead, less easily confused with what metadata is

first_field (match_info) -> context on how close the match is
second_field (metadata) -> details of the actaully matched object

naming is hard

offline side conversation: should some of this be type enforced? (follow up PR)